xayn-ai/src/coi/system.rs

```diff
@@ -40,16 +42,38 @@ pub(crate) enum CoiSystemError {
     NoMatchingDocuments,
 }
 
+/// A finite relevance score.
+#[derive(Clone, Copy, Debug, Default, Into, PartialEq, PartialOrd)]
+pub struct Relevance(f32);
+
+impl Eq for Relevance {
+    // never nan by construction
```

Comment:
no one outside of the module can construct it because the field is private. within the module we guarantee that the invariant holds. this is no different if we had an explicit constructor function.